### PR TITLE
Imputation scaling

### PR DESCRIPTION
**Pull request by kachulis**
The imputation pipeline currently does not scale past a few hundred samples.  This is due to the missing site rescue code never having been tested at larger scale.  On a 20k sample cohort, the pipeline is unrunnable (the issue is wall clock time, not compute cost.  The sample id setting task runs for a week and then the vm gets shut down).

This PR fixes the scaling issue by moving the problematic task inside scatters.  Has been tested on a 20k cohort.

### CODE REVIEW

**Comment by wjdingman1** [comment]
This looks like its failing in `GatherVcfs` not so sure about the error though... looks like its an IllegalArumentException -> [gs://broad-gotc-dev-cromwell-execution/Imputation/067905e9-33e7-446e-a081-1bd6404735ea/call-GatherVcfs/stderr](gs://broad-gotc-dev-cromwell-execution/Imputation/067905e9-33e7-446e-a081-1bd6404735ea/call-GatherVcfs/stderr)

**Comment by wjdingman1** [comment]
Im not familiar with the pipeline so I don't know if this is something on our end we need to address or what. Just let me know. thanks!

**Comment by kachulis** [comment]
@wjdingman1 yeah, this PR currently introduces some brittleness, in that it requires any contigs that are not imputed to be "after" all contigs that are imputed.  Since our tests run imputation over only chromosomes 19 and 20, this assumption is broken, and the pipeline fails.  But, I think I can make some minor changes to remove this requirement, which will be better scientifically anyway.

File: tasks/broad/ImputationTasks.wdl [hunk 1]
```diff
@@ -25,6 +25,85 @@ task CalculateChromosomeLength {
...
+}
+
+task SortChunksByContigs {
+  input {
+    File vcfs_and_indices_and_contigs_list
+    File ref_dict
+  }
+
+  command <<<
+    set -e -o p
```
Comment (nikellepetrillo) [inline tasks/broad/ImputationTasks.wdl:63]:
```suggestion
    set -e -o -p
```

Comment (nikellepetrillo) [reply tasks/broad/ImputationTasks.wdl:63]: not sure if you meant for this to be "-p" or if p was just a typo

**Comment by wjdingman1** [comment]
retest this please

File: tasks/broad/ImputationTasks.wdl (hunk 1 above)
Comment (wjdingman1) [reply tasks/broad/ImputationTasks.wdl]: think this is supposed to be pipefail?

**Comment by wjdingman1** [comment]
retest this please

**Comment by wjdingman1** [comment]
@kachulis wanted to follow up on this, looks like the changes you made allowed the pipeline to run to completion but it is failing our Plumbing test during `CompareOutputVcfs`. You can see the bucket for this task [here](https://console.cloud.google.com/storage/browser/broad-gotc-dev-cromwell-execution/VerifyImputation/c77e6352-347b-4b00-8fb4-62abc94b7322/call-CompareOutputVcfs;tab=objects?pageState=(%22StorageObjectListTable%22:(%22f%22:%22%255B%255D%22))&authuser=1&prefix=&forceOnObjectsSortingFiltering=false).

The files in question are:

truth: "gs://broad-gotc-test-storage/imputation/plumbing/truth/master/NA12878_trio/plumbing_test.vcf.gz",
test: "gs://broad-gotc-test-results/dev/imputation/plumbing/2022-04-08-11-47-29/NA12878_trio/plumbing_test.vcf.gz"

Under the hood its just doing a diff of the two files and since the test passed  the metrics verification step, it's probably safe, so if the change is within an acceptable threshold for you we can move it along.

File: tasks/broad/ImputationTasks.wdl (hunk 1 above)
Comment (kachulis) [reply tasks/broad/ImputationTasks.wdl]: haha yup :)

**Comment by kachulis** [comment]
It looks like there are some test failures related to the rna with umis pipeline; my guess is those will go away with a rebase.  But when I tried to rebase, there are conflicts in Imputation.changelog.md, and I wasn't sure how those should be resolved, so could use some help with that.

The smart test failures in the imputation pipeline are solely because of the variant id's changing, which was expected.  Other than the variant ids, the records in the test run vcf are identical to the "truth":

```
gsutil cat  gs://broad-gotc-test-storage/imputation/plumbing/truth/master/NA12878_trio/plumbing_test.vcf.gz | bcftools annotate --set-id "." | bcftools view -H | md5
1bec801d09fbb5991fdba6c1bb9c5c32

gsutil cat gs://broad-gotc-test-results/dev/imputation/plumbing/2022-04-15-09-33-09/NA12878_trio/plumbing_test.vcf.gz | bcftools annotate --set-id "." | bcftools view -H | md5
1bec801d09fbb5991fdba6c1bb9c5c32
```

**Review by wjdingman1 — APPROVED** [review]

**Comment by wjdingman1** [comment]
retest this please

**Comment by wjdingman1** [comment]
@nikellepetrillo can you add a review when you get the chance?

**Pull request merged**